File: 3075_Maximize_Happiness_of_Selected_Children/solution.py
class Solution:
    def maximumHappinessSum(self, happiness: List[int], k: int) -> int:
        happiness.sort(reverse=True)
        max_happy = 0

        for i in range(k):
            max_happy += max(0, happiness[i] - i)
        return max_happy

File: 3075_Maximize_Happiness_of_Selected_Children/solution.c
int comp(const void* a, const void* b) { return (*(int*)b - *(int*)a); }
int max(int a, int b) { return a > b ? a : b; }
long long maximumHappinessSum(int* happiness, int happinessSize, int k) {
    qsort(happiness, happinessSize, sizeof(int), comp);
    long long max_happiness = 0;
    for (int i = 0; i < k; i++) {
        max_happiness += max(0, happiness[i] - i);
    }
    return max_happiness;
}
